crates/rpc/rpc-eth-api/src/helpers/block.rs

```diff
@@ -253,19 +262,48 @@ pub trait LoadBlock: LoadPendingBlock + SpawnBlocking + RpcNodeCoreExt {
                 };
             }
 
-            let block_hash = match self
-                .provider()
-                .block_hash_for_id(block_id)
-                .map_err(Self::Error::from_eth_err)?
-            {
-                Some(block_hash) => block_hash,
-                None => return Ok(None),
-            };
+            let block = match block_id {
+                BlockId::Number(BlockNumberOrTag::Latest) => {
+                    if let Some(block) = self
+                        .cache()
+                        .latest_block_with_senders()
+                        .await
+                        .map_err(Self::Error::from_eth_err)?
+                    {
+                        Some(block)
+                    } else {
+                        // Fallback to traditional lookup if latest isn't cached
+                        match self
+                            .provider()
+                            .block_hash_for_id(block_id)
+                            .map_err(Self::Error::from_eth_err)?
+                        {
+                            Some(block_hash) => self
+                                .cache()
+                                .get_sealed_block_with_senders(block_hash)
+                                .await
+                                .map_err(Self::Error::from_eth_err)?,
+                            None => None,
+                        }
+                    }
+                }
+                _ => {
+                    let block_hash = match self
+                        .provider()
+                        .block_hash_for_id(block_id)
+                        .map_err(Self::Error::from_eth_err)?
+                    {
+                        Some(block_hash) => block_hash,
+                        None => return Ok(None),
+                    };
```

Comment:
@mattsse A bit stuck on the integration of the new function `latest_block_with_senders`, I have to add this fallback to fix the integration test but the code is very redundant

Reply:
we can look at this later, I believe we can simplify this a bit by introducing a tmp var `let mut maybe_block = None;`

and then do a `if block_id.is_latest() { maybe_block = ... }`